### PR TITLE
feature: channel argument support

### PR DESCRIPTION
**Pull request by rgrinberg**
I haven't added real support for this, but at least we recognize the common flags now so that users will get decent error messages.

### CODE REVIEW

File: ocaml-lsp-server/bin/main.ml [hunk 1]
```diff
@@ -1,16 +1,61 @@
...
-    (fun _ -> raise (Arg.Bad "anonymous arguments are not accepted"))
-    "ocamllsp";
+  in
+  let usage = "ocamllsp [ stdio ]" in
+  Arg.parse spec (fun arg -> anon_args := arg :: !anon_args) usage;
+  let anon_args = List.rev !anon_args in
+  let channel =
+    match
+      match anon_args with
+      | [] -> Ocaml_lsp_server.Stdio
```
Comment (ulugbekna) [inline ocaml-lsp-server/bin/main.ml:26]: we now only support stdio, but I think we should be checking here whether `!pipe` or `!port` is Some, right? was this omitted intentionally?

Comment (rgrinberg) [reply ocaml-lsp-server/bin/main.ml:26]: Do you mean is None? Those arguments are useless with Stdio.

Comment (rgrinberg) [reply ocaml-lsp-server/bin/main.ml]: I fixed the argument parsing a little.

**Comment by rgrinberg** [comment]
I misinterpreted how the flags were passed. This is now fixed and I've added support for all the other modes as well.

**Review by ulugbekna — APPROVED** [review]

**Pull request merged**